### PR TITLE
CB-20081 Move FMS SecurityConfig generation during creation out of request time

### PR DESCRIPTION
**Pull request by lacikaaa**
Due to low entropy it's possible that FreeIPA creation request times out when generating secrets.
To avoid such issue we are moving this part into the flow.

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/entity/SaltSecurityConfig.java [hunk 1]
```diff
@@ -83,10 +83,12 @@ public void setSaltBootSignPublicKey(String saltBootSignPublicKey) {
         this.saltBootSignPublicKey = saltBootSignPublicKey;
     }
 
+    @Deprecated
```
Comment (sodre90) [inline freeipa/src/main/java/com/sequenceiq/freeipa/entity/SaltSecurityConfig.java:86]: Can we add some hints why these are deprecated?

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java [hunk 2]
```diff
@@ -43,24 +57,41 @@ public SecurityConfig findOneByStack(Stack stack) {
...
+        }
+    }
+
     public void changeSaltPassword(Stack stack, String password) {
         SaltSecurityConfig saltSecurityConfig = Optional.ofNullable(stack.getSecurityConfig())
                 .map(SecurityConfig::getSaltSecurityConfig)
                 .orElseThrow(() -> new IllegalStateException("Stack " + stack.getResourceCrn() + " does not yet have a salt security config"));
         saltSecurityConfig.setSaltPassword(password);
-        disabledSaltSecurityConfigRepository.save(saltSecurityConfig);
+        saltSecurityConfig.setSaltPasswordVault(password);
```
Comment (sodre90) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java:94]: is it a fix for a current bug?

**Review by sodre90 — APPROVED** [review]

File: freeipa/src/main/java/com/sequenceiq/freeipa/entity/SaltSecurityConfig.java (hunk 1 above)
Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/entity/SaltSecurityConfig.java:86]: these should have been removed, I have filed a jira for this to remove them, but I would have to file some sre ticket to validate if I can remove them.

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java [hunk 3]
```diff
@@ -43,24 +57,41 @@ public SecurityConfig findOneByStack(Stack stack) {
...
                 }
-                saltSecurityConfig = disabledSaltSecurityConfigRepository.save(saltSecurityConfig);
+                saltSecurityConfig = saltSecurityConfigRepository.save(saltSecurityConfig);
                 securityConfig.setSaltSecurityConfig(saltSecurityConfig);
             }
         }
         return securityConfig;
     }
 
+    public void createIfDoesntExists(Long stackId) throws TransactionExecutionException {
```
Comment (biharitomi) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java:73]: TYPO:
'createIfDoesntExist' :)

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java (hunk 2 above)
Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java:94]: I would say, yes, this should have been added in the first place, but slipped through review

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java (hunk 3 above)
Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/SecurityConfigService.java:73]: if you don't mind I wouldn't fix this now :)

**Pull request merged**